integration-tests/smoke/fee_boosting_test.go

```diff
@@ -0,0 +1,153 @@
+			deployment.E18Mult(5),
+			big.NewInt(9e8))
+
+		state, err := ccipdeployment.LoadOnchainState(e.Env)
+		require.NoError(t, err)
+
+		allChainSelectors := maps.Keys(e.Env.Chains)
+		require.Len(t, allChainSelectors, numChains)
+
+		tokenConfig := ccipdeployment.NewTestTokenConfig(state.Chains[e.FeedChainSel].USDFeeds)
+		// Apply migration
+		output, err := changeset.InitialDeploy(e.Env, ccdeploy.DeployCCIPContractConfig{
+			HomeChainSel:   e.HomeChainSel,
+			FeedChainSel:   e.FeedChainSel,
+			ChainsToDeploy: allChainSelectors,
+			TokenConfig:    tokenConfig,
+			MCMSConfig:     ccipdeployment.NewTestMCMSConfig(t, e.Env),
+			OCRSecrets:     deployment.XXXGenerateTestOCRSecrets(),
+		})
+		require.NoError(t, err)
+		require.NoError(t, e.Env.ExistingAddresses.Merge(output.AddressBook))
+		state, err = ccipdeployment.LoadOnchainState(e.Env)
+		require.NoError(t, err)
+
+		// Ensure capreg logs are up to date.
+		ccdeploy.ReplayLogs(t, e.Env.Offchain, e.ReplayBlocks)
+
+		// Apply the jobs.
+		for nodeID, jobs := range output.JobSpecs {
+			for _, job := range jobs {
+				// Note these auto-accept
+				_, err := e.Env.Offchain.ProposeJob(ctx,
+					&jobv1.ProposeJobRequest{
+						NodeId: nodeID,
+						Spec:   job,
+					})
+				require.NoError(t, err)
+			}
+		}
+
+		return e, state, allChainSelectors
+	}
+
+	t.Run("boost needed due to WETH price increase (also covering gas price inscrease)", func(t *testing.T) {
+		e, state, chains := setupTestEnv(t, 2)
+		runFeeboostTestCase(feeboostTestCase{
+			t:            t,
+			sender:       common.LeftPadBytes(e.Env.Chains[chains[0]].DeployerKey.From.Bytes(), 32),
+			deployedEnv:  e,
+			onchainState: state,
+			initialPrices: ccipdeployment.InitialPrices{
+				LinkPrice: deployment.E18Mult(5),
+				WethPrice: deployment.E18Mult(9),
+				GasPrice:  ccipdeployment.ToPackedFee(big.NewInt(1.8e11), big.NewInt(0)),
+			},
+			priceFeedPrices: priceFeedPrices{
+				linkPrice: deployment.E18Mult(5),
+				wethPrice: big.NewInt(9.9e8),
+			},
+			sourceChain: chains[0],
+			destChain:   chains[1],
+		})
+	})
+
+	t.Run("boost needed due to LINK price decrease", func(t *testing.T) {
+		e, state, chains := setupTestEnv(t, 2)
+		runFeeboostTestCase(feeboostTestCase{
+			t:            t,
+			sender:       common.LeftPadBytes(e.Env.Chains[chains[0]].DeployerKey.From.Bytes(), 32),
+			deployedEnv:  e,
+			onchainState: state,
+			initialPrices: ccipdeployment.InitialPrices{
+				LinkPrice: deployment.E18Mult(5),
+				WethPrice: deployment.E18Mult(9),
+				GasPrice:  ccipdeployment.ToPackedFee(big.NewInt(1.8e11), big.NewInt(0)),
+			},
+			priceFeedPrices: priceFeedPrices{
+				linkPrice: big.NewInt(4.5e18),
+				wethPrice: big.NewInt(9e8),
+			},
+			sourceChain: chains[0],
+			destChain:   chains[1],
+		})
+	})
+}
+
+func runFeeboostTestCase(tc feeboostTestCase) {
+	require.NoError(tc.t, ccipdeployment.AddLane(tc.deployedEnv.Env, tc.onchainState, tc.sourceChain, tc.destChain, tc.initialPrices))
```

Comment:
should this move to test setup better?

Reply:
fixed here: https://github.com/smartcontractkit/chainlink-ccip/pull/316